SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslationManager.cpp

```diff
@@ -47,25 +48,40 @@ void SpatialVirtualWorkerTranslationManager::AuthorityChanged(const Worker_Compo
 
 	if (!bAuthoritative)
 	{
+		// A healthy server should never lose auth (we supported worker recovery, but that only happens for disconnected servers).
 		UE_LOG(LogSpatialVirtualWorkerTranslationManager, Error,
-			   TEXT("Lost authority over the translation mapping. This is not supported."));
+			   TEXT("Lost authority over the translation mapping. This should never happen."));
 		return;
 	}
 
 	const int32 ExistingTranslatorMappingCount = Translator->GetMappingCount();
+	// Fresh deployment. We need to create partition entities before we start assigning virtual workers.
 	if (ExistingTranslatorMappingCount == 0)
 	{
-		// Fresh deployment, we need to create partition entities before we start assigning virtual workers.
 		UE_LOG(LogSpatialVirtualWorkerTranslationManager, Log,
 			   TEXT("Gained authority over the VirtualWorker translation, spawning partition entities."));
 		SpawnPartitionEntitiesForVirtualWorkerIds();
 	}
+	// Previously snapshot partition auth server crashed. Need to find that server's entry in the translator mapping
+	// so we're ready to reassign the virtual worker when the replacement server starts.
+	else if (Translator->IsReady())
+	{
+		// Because we've this server has just gained authority, the local VTM state will be empty, so we copy over
+		// from the translator.
+		VirtualToPhysicalWorkerMapping = Translator->VirtualToPhysicalWorkerMapping;
+
+		UE_LOG(LogSpatialVirtualWorkerTranslationManager, Warning,
```

Comment:
Does this need to be a warning?

Reply:
Ah, nope, apologies. Unnecessarily high log levels are a consequence of my ongoing testing.